sdc/str_arr_ext.py

```diff
@@ -671,12 +671,16 @@ def construct_string_array(context, builder):
 @lower_builtin(StringArray, types.UniTuple)
 @lower_builtin(StringArray, types.Tuple)
 def impl_string_array_single(context, builder, sig, args):
-    if isinstance(args[0], types.UniTuple):
-        assert args[0].dtype == string_type
 
-    if isinstance(args[0], types.Tuple):
-        for i in args[0]:
-            assert i.dtype == string_type or i.dtype == types.StringLiteral
+    arg = args[0]
+    if isinstance(arg, (types.UniTuple, types.List)):
+        assert (arg.dtype == string_type
+                or (isinstance(arg, types.optional) and arg.type == string_type))
```

Comment:
I'm not sure, but shouldn't we check `arg.dtype` on `optional`?

Reply:
@densmirn Yes, thank you! I''l fix that.